app/controllers/discovered_hosts_controller.rb

```diff
@@ -71,6 +71,7 @@ def update
 
   def perform_update(host, success_message = nil)
     Taxonomy.no_taxonomy_scope do
+      host.build = true
```

Comment:
Please add a test that ensures `host.build` is true when we do this action or something, that way we won't get this regression again